x/protocolpool/keeper/genesis.go

```diff
@@ -56,19 +55,19 @@ func (k Keeper) InitGenesis(ctx context.Context, data *types.GenesisState) error
 		return fmt.Errorf("failed to set last balance: %w", err)
 	}
 
-	totalToBeDistributed := math.ZeroInt()
+	totalToBeDistributed := sdk.NewCoins()
 	for _, distribution := range data.Distributions {
-		totalToBeDistributed = totalToBeDistributed.Add(distribution.Amount)
+		totalToBeDistributed = totalToBeDistributed.Add(distribution.Amount.Amount...)
 		if err := k.Distributions.Set(ctx, *distribution.Time, distribution.Amount); err != nil {
 			return fmt.Errorf("failed to set distribution: %w", err)
 		}
 	}
 
 	// sanity check to avoid trying to distribute more than what is available
-	if data.LastBalance.LT(totalToBeDistributed) {
-		return errors.New("total to be distributed is greater than the last balance")
-	}
 
+	if data.LastBalance.Amount.IsAnyGT(totalToBeDistributed) || !totalToBeDistributed.DenomsSubsetOf(data.LastBalance.Amount) {
```

Comment:
You are right about the inverted comparison, but I still need `DenomsSubsetOf`, see:

```
// IsAnyGT returns true iff for any denom in coins, the denom is present at a
// greater amount in coinsB.
//
// e.g.
// {2A, 3B}.IsAnyGT{A} = true
// {2A, 3B}.IsAnyGT{5C} = false
// {}.IsAnyGT{5C} = false
// {2A, 3B}.IsAnyGT{} = false
```

This case `{}.IsAnyGT{5C} = false` shows that if totalToBeDistributed has 1foo but LastBalance contains `1bar`, it will be false.